internal/restorer/restorer.go

```diff
@@ -571,10 +623,16 @@ func (res *Restorer) withOverwriteCheck(ctx context.Context, node *restic.Node,
 	var matches *fileState
 	updateMetadataOnly := false
 	if node.Type == restic.NodeTypeFile && !isHardlink {
-		// if a file fails to verify, then matches is nil which results in restoring from scratch
-		matches, buf, _ = res.verifyFile(ctx, target, node, false, res.opts.Overwrite == OverwriteIfChanged, buf)
-		// skip files that are already correct completely
-		updateMetadataOnly = !matches.NeedsRestore()
+		// Submit verification task without waiting
+		err = res.submitVerifyTask(ctx, target, node, location, false, res.opts.Overwrite == OverwriteIfChanged, cb)
+		if err != nil {
+			debug.Log("concurrent verify submit failed for %s, falling back to sync: %v", target, err)
```

Comment:
This is non-sense. Either the worker goroutines work or the whole operation is doomed anyways.